### PR TITLE
In Archetype Delete, call PatchMeta. Fixes https://github.com/outfox/fennecs/issues/23

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by PennyMew** [comment]
I didn't see contributor guidelines, I've only been looking at fennecs source code for a few hours, and I don't regularly contribute to open source, so please let me know if there's something I should do differently.

There's one TODO comment that needs to be addressed. I wasn't sure where to put the new unit test.

File: fennecs/Archetype.cs [hunk 1]
```diff
@@ -167,6 +175,14 @@ public void Truncate(int maxEntityCount)
         IdentityStorage.Delete(Count - excess, excess);
     }
 
+    /// <summary>
+    /// Update Entity metas with new Row information. Call to update the Meta Row information
+    /// after moving Entities/Component data within the Storages. This ensures that the Entity
+    /// Meta within the World will have the correct Row to access the Entity's associated data
+    /// within the Archetype.
+    /// </summary>
+    /// <param name="entry">If `count` param is nonzero, `entry` must be less than `this.Count`.</param>
```
Comment (PennyMew) [inline fennecs/Archetype.cs:184]: Technically should replace "nonzero" with "greater than zero"

**Comment by thygrrr** [comment]
Holy moly, indeed. Let me verify this.

**Comment by thygrrr** [comment]
Thank you so very much for providing this vital fix.

**Pull request merged**